lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1689,6 +1689,8 @@ def main():
                 # has to be poweredoff first
                 pyv.set_powerstate(vm, 'poweredoff', module.params['force'])
             result = pyv.remove_vm(vm)
+        elif module.params['snapshot_op']:
```

Comment:
yes i think it should be pulled outside but the fix will break module idempotency :( please just fix either the test or push the new module (not needed there)